apt/minimization/minimizer.py

```diff
@@ -256,6 +256,7 @@ def fit(self, X: Optional[DATA_PANDAS_NUMPY_TYPE] = None, y: Optional[DATA_PANDA
         # Going to fit
         # (currently not dealing with option to fit with only X and y and no estimator)
         if self.estimator and dataset and dataset.get_samples() is not None and dataset.get_labels() is not None:
+            dtype = dataset.get_samples().dtype
```

Comment:
Can't the data be a pandas? Then this would fail.

Reply:
ArrayDataset always returns a numpy array regardless of the original type of the data that was supplied to it.